docs/architecture/app-class-and-template.md

```diff
@@ -0,0 +1,160 @@
+# ApplicationClass and ApplicationTemplate
+
+Managing [Applications](./../terminology.md#Application) can be passed on some external system, to remove the burden of integration with Compass from Applications.
+Because of that [IntegrationSystem](./../terminology.md#Integration-System) and ApplicationClass types were introduced.
+
+To increase the reusability of the application configuration, we introduce ApplicationTemplate.
+The diagram below describes when to use ApplicationClass or ApplicationTemplate.
+
+![](./assets/app-vs-template-vs-class.svg)
+
+## Terminology
+
+- **ApplicationClass** represents an object for creating Applications. When you create an application from it, the IntegrationSystem
+dynamically populates definition.
```

Comment:
populates -> updates?
and you can mention here controlling some fields, such as API or EventAPI definitions